src/services/statistic/index.js

```diff
@@ -7,129 +7,192 @@ const userModel = require('../user/model');
+		},
+		{
+			name: 'accounts',
+			promise: accountModel.countDocuments(),
+			model: accountModel.find(),
+		},
+		{
+			name: 'teams',
+			promise: teamsModel.teamsModel.countDocuments(),
+			model: teamsModel.teamsModel.find(),
+		},
+		{
+			name: 'lessons',
+			promise: lessonModel.countDocuments(),
+			model: lessonModel.find(),
+		},
+		{
+			name: 'files_directories',
+			promise: FileModel.countDocuments(),
+			model: FileModel.find(),
+		},
+		{
+			name: 'files_sizes',
+			promise: FileModel.aggregate([
+				{
+					$bucketAuto: {
+						groupBy: '$size',
+						buckets: 5,
+					},
 				},
-			},
-		]),
-		model: FileModel.find(),
-	},
-	{
-		name: 'files/types',
-		promise: FileModel.aggregate([
-			{
-				$group: {
-					_id: '$type',
-					total_files_per_type: { $sum: 1 },
+			]),
+			model: FileModel.find(),
+		},
+		{
+			name: 'files_types',
+			promise: FileModel.aggregate([
+				{
+					$group: {
+						_id: '$type',
+						total_files_per_type: { $sum: 1 },
+					},
 				},
-			},
-		]),
-		model: FileModel.find(),
-	},
-];
+			]),
+			model: FileModel.find(),
+		},
+	];
+
+	if (!schoolId) {
+		promises = [...promises, ...promisesWithoutSchoolIds];
+	}
 
-const fetchStatistics = () => {
+	return promises;
+};
+
+const fetchStatistics = (schoolId) => {
 	const statistics = {};
 
-	return Promise.all(
-		promises.map((p) => p.promise.exec().then((res) => {
-			statistics[p.name] = res;
-			return res;
-		})),
-	).then(() => statistics);
+	return Promise.all(getPromises(schoolId).map((p) => p.promise.exec().then((res) => {
+		statistics[p.name] = res;
+		return res;
+	}))).then((_) => statistics);
 };
 
 class StatisticsService {
 	constructor() {
 		this.docs = swaggerDocs.statisticsService;
 	}
 
-	find() {
-		return fetchStatistics()
-			.then((statistics) => statistics);
+	setup(app) {
+		this.app = app;
+	}
+
+	async find({
+		query, params, account, app,
+	}) {
+		let statistics;
+		// TODO: Promise.all
+		const hasViewMySchoolStatsPermission = await hasPermissionNoHook(
+			this, account.userId, PERMISSIONS.VIEW_MYSCHOOL_STATS,
+		);
+		const hasViewGlobalStatsPermission = await hasPermissionNoHook(
+			this, account.userId, PERMISSIONS.VIEW_GLOBAL_STATS,
+		);
+		if (query.school === 'myschool' && hasViewMySchoolStatsPermission) {
+			statistics = await fetchStatistics(account.schoolId);
+		} else if (hasViewGlobalStatsPermission) {
+			statistics = await fetchStatistics();
+		}
+		return statistics;
 	}
 
 	get(id, params) {
-		return _.find(promises, { name: id })
+		return _.find(getPromises(), { name: id })
 			.model.select({ createdAt: 1 })
 			.exec()
 			.then((generic) => {
-				const stats = generic.map((gen) => moment(gen.createdAt).format('YYYY-MM-DD'));
+				const stats = generic.map((gen) => moment(gen.createdAt).toISOString());
 
 				const counts = {};
-				stats.forEach((x) => {
-					counts[x] = (counts[x] || 0) + 1;
+				stats.forEach((x) => { 
+					counts[x] = (counts[x] || 0) + 1; 
```

Comment:
Codacy found an issue: [Trailing spaces not allowed.](https://app.codacy.com/gh/hpi-schul-cloud/schulcloud-server/pullRequest?prid=6045807)